### DISCUSSION TITLE
Can I specify secrets when creating a Code Sandbox via POST API?

### DISCUSSION THREAD

**Discussion by petrbrzek** [codesandbox/codesandbox-client#5732] [Q&A] [answered]
Hey 👋 ,

I'm using the CS Container in our docs for API examples and I wanted to dynamically generate a codesandbox with baked API Token, so users work with their data. It's probably not possible right now? I don't see anything about it here https://codesandbox.io/docs/api.

Thanks.

**Comment by andrasbacsai** [comment, marked as answer]
Hey,

You can use our `Define API` to dynamically define a sandbox through our API: https://codesandbox.io/docs/api/#define-api

Here is a payload for defining a simple container sandbox:
```json
{
    "files": {
        "package.json": {
            "content": {
                "scripts": {
                    "start": "./node_modules/node/node_modules/.bin/node ./node_modules/nodemon/bin/nodemon.js ./src/index.js"
                },
                "dependencies": {
                    "node": "14.2.0"
                },
                "devDependencies": {
                    "nodemon": "^1.18.4"
                }
            }
        },
        "src/index.js": {
            "content": "console.log('helloa')"
        },
        "index.html": {
            "content": "<html>hello</html>"
        }
    }
}
```

**Reply by petrbrzek** [reply]
Thanks, but I was asking about specifying secrets via API.
